Final Legal Chatbot Files/scraping_load.py

```
@@ -0,0 +1,243 @@
+    candidate_docs = dedupe_documents(docs_lt + docs_law)
+    ranked_docs, ranked_scores = rerank_documents(query, candidate_docs)
+
+    final_docs, final_scores = threshold_reranked_docs(
+        ranked_docs, ranked_scores,
+        min_score=RERANK_MIN_SCORE,
+        top_k=max(k, MIN_LT_RESULTS + MIN_LAW_RESULTS),
+    )
+
+    if role and role != "general":
+        role_docs, non_role_docs = [], []
+        for doc, score in zip(final_docs, final_scores):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)